test/Datadog.Trace.ClrProfiler.IntegrationTests/AdoNet/MicrosoftDataSqlClientTests.cs

```diff
@@ -0,0 +1,66 @@
+#if !NET452
+using Datadog.Core.Tools;
+using Datadog.Trace.TestHelpers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests.AdoNet
+{
+    public class MicrosoftDataSqlClientTests : TestHelper
+    {
+        public MicrosoftDataSqlClientTests(ITestOutputHelper output)
+            : base("Microsoft.Data.SqlClient", output)
+        {
+            SetServiceVersion("1.0.0");
+        }
+
+        [Theory]
+        [MemberData(nameof(PackageVersions.MicrosoftDataSqlClient), MemberType = typeof(PackageVersions))]
+        [Trait("Category", "EndToEnd")]
+        [Trait("RunOnWindows", "True")]
+        public void SubmitsTracesWithNetStandard(string packageVersion)
+        {
+            // Note: The automatic instrumentation currently does not instrument on the generic wrappers
+            // due to an issue with constrained virtual method calls. This leads to an inconsistency where
+            // the .NET Core apps generate 4 more spans than .NET Framework apps (2 ExecuteReader calls *
+            // 2 interfaces: IDbCommand and IDbCommand-netstandard).
+            // Once this is fully supported, this will add another 2 complete groups for all frameworks instead
+            // of 4 extra spans on net461 and netcoreapp2.0+
+#if NET452
+            var expectedSpanCount = 49; // 7 queries * 7 groups
+#elif NET461
+            var expectedSpanCount = 73; // 7 queries * 11 groups - ?
```

Comment:
So I realized these numbers are 4 fewer than the System.Data.SqlClient tests because the ExecuteReaderAsync method is not getting instrumented. This is called twice in the SqlCommand->async sub-trace and twice in the SqlCommand->async_with_cancellation sub-trace. I'll see if I can figure it out 🔍

Reply:
I found the issue and fixed it in 5f3b269cab32009d1eaf663f2383e90012ae8fa4 . The integration method wasn't being called because the return type name was incorrect. I've also updated the span counts now that we're generating a few more spans. It now matches SystemDataSqlClientTests.

Reply:
Wow, thanks!